src/encoder.rs

```diff
@@ -20,7 +20,23 @@ pub type Result<T> = result::Result<T, EncodingError>;
 #[derive(Debug)]
 pub enum EncodingError {
     IoError(io::Error),
-    Format(Cow<'static, str>),
+    Format(FormatError),
```

Comment:
I'm only a bit puzzled by the asymmetry of variants between `DecodingError` and `EncodingError`. I'm not sure if we should introduce at least `ParameterError` here—and keep it unused for now. Maybe a method to manually close the png (i.e. write and IEND chunk) by `&mut self` could be useful to avoid fallible writes in Drop. After it, all other methods should return that error I think.